test/functional/cursor.test.js

```diff
@@ -556,14 +556,15 @@ describe('Cursor', function() {
           }
 
           function finished() {
-            collection.find(function(err, cursor) {
-              test.equal(null, err);
+            (function() {
```

Comment:
There are many nested `cursor` functions. I found rather then `cursorOne`, `cursorTwo`, `cursorThree` it would be easier to encapsulate in IIFEE rather then overwrite the same variable or renaming.

Reply:
When the cursor is only having one test on it, it might be more readable to skip the assignment entirely. E.g. 
```
test.throws(
  function() {
    collection.find().each();
  }
);
```

Reply:
@emadum Yeah totally. Many cases however the cursor is used later on in the test.

Here's an example. 

```
const cursor = collection.find().batchSize(1);
cursor.next(function(err, item) {
    test.equal(null, err);
    test.equal(1, item.a);

    // Need to close cursor, since it was not exhausted,
    // and implicit session is still open
    cursor.close();
    client.close(done);
});
```
For consistency I like assigning it. `const cursor = ...`

@mbroadst I removed any IIIFE that is not necessary. I'm limited it to only cases where there are many cursors in one test. How do we feel about that? Totally get if it's not a good solution for this scoping issue.